x509/codesigning_cert_validations.go

```diff
@@ -51,22 +51,18 @@ func ValidateCodeSigningCertChain(certChain []*x509.Certificate, signingTime *ti
 			return signedTimeError
 		}
 		if i == len(certChain)-1 {
-			selfSigned, selfSignedError := isSelfSigned(cert)
-			if selfSignedError != nil {
-				return fmt.Errorf("root certificate with subject %q is invalid or not self-signed. Certificate chain must end with a valid self-signed root certificate. Error: %v", cert.Subject, selfSignedError)
-			}
-			if !selfSigned {
-				return fmt.Errorf("root certificate with subject %q is not self-signed. Certificate chain must end with a valid self-signed root certificate", cert.Subject)
+			if err := validateSelfSignedCert(cert, false); err != nil {
+				return err
```

Comment:
After reading through the code again, I think we'd better keep the original `isSelfSigned` logic:
It's used for cert chain with length gt than 1 and the target cert can be leaf/intermediate/root. So it's not assuming the target cert is a root certificate. In fact, `isSelfSigned` does not have any cert chain context, it only checks self-signed.

Reply:
Updated.